src/parser/tokenizer.ts

```diff
@@ -234,6 +233,17 @@ export class Tokenizer {
     return new IdentifierToken(this.input, begin, this.p, this.file)
   }
 
+  readTagName (): string {
+    this.skipBlank()
+    const begin = this.p
+    // Handle inline comment tags
+    if (this.input[this.p] === '#') {
+      return this.input.slice(begin, ++this.p)
+    }
```

Comment:
can we call `readIdentifier` here?

Reply:
Can do, after the check for `#`. '#' being an invalid character for identifiers.

I was expecting a minor performance improvement by avoiding the creation of an `IdentifierToken` for every tag name, as `readIdentifier` does. Both `TagToken` and `LiquidTagToken` were immediately calling `getText()` on the `IdentifierToken` to get the string. I've not benchmarked it though.

Reply:
Benchmarking showed no discernible performance improvement. Refactored accordingly in 200fb36.

Reply:
Makes sense, I think it’s OK either way.